(BKR-1462) Another Ubuntu 18.04 check to allow insecure repo use

ScottGarman opened the pull request.

speedofdark commented:
No permissions to review, but (corpsethumb)...

MikaelSmith commented on lib/beaker-puppet/install_utils/foss_utils.rb:
```diff
@@ -1035,6 +1035,10 @@ def install_repo_configs_from_url(host, repo_config_url, copy_dir = nil)
           end
           scp_to( host, repo, to_path )
 
+          if host['platform'] =~ /ubuntu-18.04/
```
This appears to make the previous change redundant. The puppet-agent install helper, after setting this, immediately calls code that calls install_repo_configs_from_url.

MikaelSmith replied:
Tying it to apt-get update makes sense. We'll add a repo without signing here, the only other apt-get update corresponds to public repos which should be signed.

MikaelSmith replied:
The most relevant place might be in install_puppetlabs_dev_repo, since that's the specific type of repo we expect to be unsigned.

MikaelSmith commented:
I'm retesting. @caseywilliams what was the other pipeline that needed the previous fix?

ScottGarman commented:
Ok, as @MikaelSmith suggested, I've moved the apt config change into install_puppetlabs_dev_repo, and I've also reverted the commit where the first pass attempt was made. Please test this out and let me know if it works.

caseywilliams commented:
Also still works for puppet-agent acceptance!

ScottGarman commented:
I've tried to fix the failing PR tests by removing the revert commit and just including the change in this single commit. No difference. Sounds like this is ready for merge now.

MikaelSmith commented:
Now green.

MikaelSmith commented on lib/beaker-puppet/install_utils/foss_utils.rb:
```diff
@@ -1080,6 +1080,11 @@ def install_puppetlabs_dev_repo ( host, package_name, build_version,
           dev_builds_url = protocol + hostname
           dev_builds_url = opts[:dev_builds_url] if variant =~ /^(fedora|el|centos)$/
 
+          if variant =~ /^ubuntu$/ && version.split('.').first.to_i >= 18
```
This is now redundant, since we immediately call `install_repo_configs` -> `install_repo_configs_from_url`.

MikaelSmith replied:
I still really dislike the redundancy here. And I think a spec test covering this would be useful (at the level of: this function ensures that we allow unsigned packages on Ubuntu 18.04).

caseywilliams commented on lib/beaker-puppet/install_utils/foss_utils.rb:
```diff
@@ -1035,6 +1035,11 @@ def install_repo_configs_from_url(host, repo_config_url, copy_dir = nil)
+          variant, version, arch, codename = host['platform'].to_array
+          if variant =~ /^ubuntu$/ && version.split('.').first.to_i >= 18
+            # Allow the use of unsigned repos with Ubuntu 18.04+
+            on host, "echo 'Acquire::AllowInsecureRepositories \"true\";' > /etc/apt/apt     .conf.d/90insecure"
```
Ah, I should have noticed this before running the tests, but the spaces in the filename are stopping this from working -- this line is applied in the correct tests though.

ScottGarman commented:
Apologies for that, PR updated.

ScottGarman commented:
FWIW I've filed a ticket (BKR-1468) to eventually fix this duplication.
